migration/src/m0000641_update_product_status.rs

```diff
@@ -0,0 +1,130 @@
+use sea_orm_migration::prelude::*;
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[async_trait::async_trait]
+#[allow(deprecated)]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .alter_table(
+                Table::alter()
+                    .table(ProductStatus::Table)
+                    .add_column(ColumnDef::new(ProductStatus::Component).string())
+                    .to_owned(),
+            )
+            .await?;
+
+        // Transform data: populate `component` from `base_purl`
+        let update_sql = r#"
+            UPDATE product_status
+            SET component =
+                (SELECT namespace || '/' || name FROM base_purl WHERE base_purl.id = product_status.base_purl_id)
+            WHERE base_purl_id IS NOT NULL;
+        "#;
+        manager
+            .get_connection()
+            .execute_unprepared(update_sql)
+            .await?;
+
+        // Delete original `purl_status` entries now that data has been migrated
+        let delete_statuses_sql = r#"
+            DELETE FROM purl_status
+            WHERE base_purl_id IN (
+                SELECT id FROM base_purl
+                WHERE id IN (SELECT DISTINCT base_purl_id FROM product_status WHERE base_purl_id IS NOT NULL)
+                AND TYPE = 'generic'
```

Comment:
for my own education ... why the TYPE='generic' filter is relevant ?

Reply:
This is how previous logic stored those "fake" purls, so basically `io.quarkus/quarkus-vertx-http` was stored as `generic://io.quarkus/quarkus-vertx-http`. So we need to clean those up